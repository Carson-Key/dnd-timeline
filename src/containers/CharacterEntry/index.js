import React, { Component } from 'react'
import './index.css'
import {
  InputGroup,
  InputGroupAddon,
  Input,
  Button,
  Container,
  Row,
  Col,
  Jumbotron,
  Alert
} from 'reactstrap';
import Time from '../Time'
import { filterTimeLine } from '../../utilities/filterTimeLine.js'

class CharacterEntry extends Component {
  constructor(props) {
    super(props);
    this.state = {
      error: ""
    }
  }

  displayError() {
    if (this.state.error !== "") {
      return (
        <Alert color="danger">
          Please Enter a valid player name
        </Alert>
      )
    } else {
      return
    }
  }

  render() {
    return (
      <Container>
        {this.displayError()}
        <br />
        <Row>
          <Col>
            <Jumbotron fluid>
              <Container fluid>
                <center><h1 className="display-3">Timeline</h1></center>
              </Container>
            </Jumbotron>
          </Col>
        </Row>
        <Row className="centered">
          <Col>
            <p>What is Your Player's Name?</p>
            <InputGroup>
              <Input
                placeholder="Cade Goodbarrel"
                value={this.state.playerName}
                onChange={this.props.onChange}
                onKeyPress={this.props.onKeyPress}
              />
              <InputGroupAddon addonType="append">
                <Button onClick={this.props.enterFunc}>Enter</Button>
              </InputGroupAddon>
            </InputGroup>
          </Col>
        </Row>
      </Container>
    )
  }
}

export default CharacterEntry
