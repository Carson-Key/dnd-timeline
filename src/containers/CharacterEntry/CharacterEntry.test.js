import React from "react";
import { render } from "@testing-library/react";
import CharacterEntry from "./index.js";
import renderer from 'react-test-renderer';

describe('Render CharacterEntry ->', () => {

  test('with empty playerObject prop', () => {
    const component = renderer.create(
      <CharacterEntry
        playerObject={
          {characters: {}}
        }
      />,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('with one character', () => {
    const component = renderer.create(
      <CharacterEntry
        playerObject={
          {characters: {
            "cade": {
                "name": "Cade Goodbarrel",
                "characterKey": "cade",
                "timeline": {
                  "events": {}
                }
              }
          }}
        }
      />,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('with multiple characters', () => {
    const component = renderer.create(
      <CharacterEntry
        playerObject={
          {
            characters: {
              "cade": {
                "name": "Cade Goodbarrel",
                "characterKey": "cade",
                "timeline": {
                  "events": {}
                }
              },
              "cade2": {
                "name": "Cade Goodbarrel",
                "characterKey": "cade2",
                "timeline": {
                  "events": {}
                }
              }
            }
          }
        }
      />,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

})
