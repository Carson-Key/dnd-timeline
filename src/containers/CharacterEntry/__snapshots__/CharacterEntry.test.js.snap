// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render CharacterEntry -> with empty playerObject prop 1`] = `
<div
  className="container"
>
  <br />
  <div
    className="row"
  >
    <div
      className="col"
    >
      <center>
        <p>
          Pick a character
        </p>
        <br />
      </center>
    </div>
  </div>
</div>
`;

exports[`Render CharacterEntry -> with multiple characters 1`] = `
<div
  className="container"
>
  <br />
  <div
    className="row"
  >
    <div
      className="col"
    >
      <center>
        <p>
          Pick a character
        </p>
        <br />
      </center>
      <center>
        <button
          aria-label={null}
          className="btn btn-secondary"
          onClick={[Function]}
          type="button"
        >
          Cade Goodbarrel
        </button>
        <br />
        <br />
      </center>
      <center>
        <button
          aria-label={null}
          className="btn btn-secondary"
          onClick={[Function]}
          type="button"
        >
          Cade Goodbarrel
        </button>
        <br />
        <br />
      </center>
    </div>
  </div>
</div>
`;

exports[`Render CharacterEntry -> with one character 1`] = `
<div
  className="container"
>
  <br />
  <div
    className="row"
  >
    <div
      className="col"
    >
      <center>
        <p>
          Pick a character
        </p>
        <br />
      </center>
      <center>
        <button
          aria-label={null}
          className="btn btn-secondary"
          onClick={[Function]}
          type="button"
        >
          Cade Goodbarrel
        </button>
        <br />
        <br />
      </center>
    </div>
  </div>
</div>
`;
